File: src/main/java/io/automatizacionconvocatorias/sdk/Template.java
package io.automatizacionconvocatorias.sdk;

import java.util.Objects;

/**
 * Represents a template in the Convocatorias marketplace.
 */
public class Template {
    private String id;
    private String name;
    private String category;
    private String content;

    // Constructors
    public Template() {}

    public Template(String id, String name, String category, String content) {
        this.id = id;
        this.name = name;
        this.category = category;
        this.content = content;
    }

    // Getters and Setters
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Template template = (Template) o;
        return Objects.equals(id, template.id) &&
                Objects.equals(name, template.name) &&
                Objects.equals(category, template.category) &&
                Objects.equals(content, template.content);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name, category, content);
    }

    @Override
    public String toString() {
        return "Template{" +
                "id='" + id + '\'' +
                ", name='" + name + '\'' +
                ", category='" + category + '\'' +
                ", content='" + content + '\'' +
                '}';
    }
}

File: src/Convocatorias.Sdk/Template.cs
using System.Text.Json.Serialization;

namespace Convocatorias.Sdk
{
    /// <summary>
    /// Represents a template in the Convocatorias marketplace.
    /// </summary>
    public class Template
    {
        public string Id { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public string Content { get; set; } = string.Empty;
        public string? Description { get; set; }
        public int Version { get; set; } = 1;
        public bool IsActive { get; set; } = true;
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }

        public Template() { }

        public Template(string id, string name, string category, string content)
        {
            Id = id;
            Name = name;
            Category = category;
            Content = content;
        }

        public Template(string id, string name, string category, string content, string? description) : this(id, name, category, content)
        {
            Description = description;
        }

        public override bool Equals(object? obj)
        {
            return obj is Template template &&
                   Id == template.Id &&
                   Name == template.Name &&
                   Category == template.Category &&
                   Content == template.Content;
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(Id, Name, Category, Content);
        }

        public override string ToString()
        {
            return \$"Template{{Id='{Id}', Name='{Name}', Category='{Category}'}}";
        }
    }
}
